src/box/replication.cc

```diff
@@ -714,6 +841,11 @@ replicaset_update(struct applier **appliers, int count, bool keep_connect)
 			replica_delete(replica);
 		}
 	}
+	/*
+	 * Remember how many appliers are connected to check
+	 * replicaset_sync_quorum later on.
+	 */
+	replicaset_set_sync_quorum();
```

Comment:
So if some appliers were not connected here but will reconnect later, we still do not include them into the sync?

Reply:
Yep. But we wait for everyone to connect for a `replication_connect_timeout`. Even on local recovery and reconfiguration.